WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerActivity.kt

```diff
@@ -49,8 +129,96 @@ class StoryComposerActivity : ComposeLoopFrameActivity(), SnackbarProvider, Medi
                 this,
                 MediaBrowserType.WP_STORIES_MEDIA_PICKER,
                 site,
-                null // TODO obtain the local PostId when integrating with FluxC model
+                getTestPost(postStore, site!!).id // TODO obtain the local PostId when integrating with FluxC model
                 // mEditPostRepository.id
         )
     }
+
+    override fun providerHandlesOnActivityResult(): Boolean {
+        // lets the super class know we're handling media picking OnActivityResult
+        return true
+    }
+
+    fun handleMediaPickerResult(data: Intent) {
+        // TODO move this to EditorMedia
+        val ids = ListUtils.fromLongArray(
+                data.getLongArrayExtra(
+                        MediaBrowserActivity.RESULT_IDS
+                )
+        )
+        if (ids == null || ids.size == 0) {
+            return
+        }
+
+        editorMedia.addExistingMediaToEditorAsync(WP_MEDIA_LIBRARY, ids)
+    }
+
+    private fun startObserving() {
+        editorMedia.uiState.observe(this,
+                Observer { uiState: AddMediaToPostUiState? ->
+                    if (uiState != null) {
+                        updateAddingMediaToEditorProgressDialogState(uiState.progressDialogUiState)
+                        if (uiState.editorOverlayVisibility) {
+                            showLoading()
+                        } else {
+                            hideLoading()
+                        }
+                    }
+                }
+        )
+        editorMedia.snackBarMessage.observe(this,
+                Observer<Event<SnackbarMessageHolder>> { event: Event<SnackbarMessageHolder?> ->
+                    val messageHolder = event.getContentIfNotHandled()
+                    if (messageHolder != null) {
+                        WPSnackbar
+                                .make(
+                                        findViewById(id.editor_activity),
+                                        messageHolder.messageRes,
+                                        Snackbar.LENGTH_SHORT
+                                )
+                                .show()
+                    }
+                }
+        )
+        editorMedia.toastMessage.observe(this,
+                Observer<Event<ToastMessageHolder>> { event: Event<ToastMessageHolder?> ->
+                    val contentIfNotHandled = event.getContentIfNotHandled()
+                    contentIfNotHandled?.show(this)
+                }
+        )
+    }
+
+    // EditorMediaListener
+    override fun appendMediaFiles(mediaFiles: Map<String, MediaFile>) {
```

Comment:
As mentioned above, we might want to consider moving all this logic into a VM. Wdyt?